### PR TITLE
feat(apps): very dodgy frontend apps implementation

### PR DESCRIPTION
**Pull request by mariusandra**
**Warning: this is dodgy as hell**

## Problem

We want plugins to show up in the sidebar.

## Changes

- Adds a text field `frontend` to the `Plugin` model.
- Patches `pluginLogic` to detect code in this field
- If so, it `eval`s the code, and adds it to the sidebar if it exports a `sidebar` object
- Adds a route, which renders the component given

Here's a demo `frontend.tsx` for a plugin:

```tsx
import React from "react";

export const sidebar = {
  title: "Demo App",
  component: function Sidebar(): JSX.Element {
    return <div>Sidebar plugin content</div>;
  },
};
```

This is transpiled through `tsc` into this `frontend.js`:

```js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.sidebar = void 0;
const react_1 = __importDefault(require("react"));
exports.sidebar = {
    title: "Demo App",
    component: function Sidebar() {
        return react_1.default.createElement("div", null, "Sidebar plugin content");
    },
};
//# sourceMappingURL=frontend.js.map
```

And then that gets loaded by the browser:

image

A real integration will require integrating `esbuild` here. Totally doable, just more work.

We also need to export a nice set of components for app builders to use. Plus kea and a few other libraries. More work, but, again, doable.

## How did you test this code?

- By cutting as many corners as possible

### CODE REVIEW

File: frontend/src/scenes/insights/utils.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ export function extractObjectDiffKeys(
 }
 
 export function findInsightFromMountedLogic(
-    insightShortId: InsightShortId,
+    insightShortId: InsightShortId | string,
```
Comment (Twixes) [inline frontend/src/scenes/insights/utils.ts:88]: Dodgy

File: frontend/src/scenes/insights/AdHocInsight.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+let uniqueMemoizedIndex = 0
+export function AdHocInsight({
+    filters,
+    style,
+}: {
+    filters: Partial<FilterType>
+    style: React.CSSProperties
+}): JSX.Element {
+    const pageKey = useMemo(() => `filter-${uniqueMemoizedIndex++}`, [])
```
Comment (Twixes) [inline frontend/src/scenes/insights/AdHocInsight.tsx:16]: Kinda dodgy to rely on this as `memo` isn't guaranteed to hold for re-renders, and the key changing could mess up what state the logic uses.

File: frontend/src/scenes/plugins/pluginsLogic.ts [hunk 3]
```diff
@@ -750,4 +777,42 @@ export const pluginsLogic = kea<pluginsLogicType<PluginForm, PluginSection, Plug
...
+                            } else if (module === 'kea') {
+                                return allKea
+                            } else if (module === '@posthog/apps-common') {
+                                return { AdHocInsight: AdHocInsight, LemonButton: LemonButton, LemonRow: LemonRow }
+                            } else {
+                                throw new Error(`Can not import from unknown module "${module}"`)
+                            }
+                        }
+                        // TODO: less YOLO
+                        eval(`${plugin.frontend}`)
```
Comment (Twixes) [inline frontend/src/scenes/plugins/pluginsLogic.ts:803]: Dodgy AF

**Review by Twixes — COMMENTED** [review]
Dodgy code, but cool results. From the product perspective though I agree with Marcus that while in the long term this seems to have significant potential, short-term it's more valuable for us to focus on the quality of our existing product.

**Comment by mariusandra** [comment]
Dodging this and continuing in https://github.com/PostHog/posthog/pull/9618

**Pull request closed without merging**